### PR TITLE
LibArchive+Utilities: Port ZipOutputStream and zip(1) to `Core::Stream`

### PR DESCRIPTION
**Pull request by krkk**

### CODE REVIEW

File: Userland/Libraries/LibArchive/Zip.cpp [hunk 1]
```diff
@@ -141,10 +141,11 @@ void ZipOutputStream::add_member(ZipMember const& member)
         .extra_data = nullptr,
         .compressed_data = member.compressed_data.data(),
     };
-    local_file_header.write(m_stream);
+    TRY(local_file_header.write(*m_stream));
+    return {};
```
Comment (alimpfard) [inline Userland/Libraries/LibArchive/Zip.cpp:145]:
```suggestion
    return local_file_header.write(*m_stream);
```

File: Userland/Libraries/LibArchive/Zip.cpp [hunk 2]
```diff
@@ -189,7 +190,8 @@ void ZipOutputStream::finish()
         .comment_length = 0,
         .comment = nullptr,
     };
-    end_of_central_directory.write(m_stream);
+    TRY(end_of_central_directory.write(*m_stream));
+    return {};
```
Comment (alimpfard) [inline Userland/Libraries/LibArchive/Zip.cpp:194]: same?

File: Userland/Utilities/zip.cpp [hunk 3]
```diff
@@ -76,10 +73,11 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         Crypto::Checksum::CRC32 checksum { file_buffer.bytes() };
         member.crc32 = checksum.digest();
         member.is_directory = false;
-        zip_stream.add_member(member);
+        TRY(zip_stream.add_member(member));
+        return {};
```
Comment (alimpfard) [inline Userland/Utilities/zip.cpp:77]: And dis.

**Review by alimpfard — COMMENTED** [review]
LGTM, but have some minor comments you can ignore if you want:

**Comment by krkk** [comment]
note to the future self: commit category is wrong (LibCompress → LibArchive)

**Comment by krkk** [comment]
changes since last push:
- applied suggested changes
- fixed commit category
- removed now unused `OutputStream& operator<<` helper function for ZipCompressionMethods

**Pull request merged**